src/util/address.rs

```diff
@@ -870,6 +890,31 @@ impl fmt::Debug for Address {
     }
 }
 
+/// decode address to Bech32 u8 byte array
+fn bech32_decode(address: &Address) -> Result<Vec<u8>, Error> {
+    match bech32::decode(&address.to_string()) {
```

Comment:
No need for a `match` statement here as `?` can be used to convert and return the error since `address::Error` implements `From<bech32::Error>`.

Reply:
Replacing with
```rust
/// decode address to Bech32 u8 byte array
fn bech32_decode(address: &Address) -> Result<Vec<u8>, Error> {
    let (_, u5_vec, _) = bech32::decode(&address.to_string())?;
    Ok(bech32::convert_bits(&u5_vec[1..], 5, 8, true)?)
}
```